app.py

```diff
@@ -541,11 +544,31 @@ def remove_file(response):
     return send_file("machine_files/" + filename + "." + extension, as_attachment=True)
 
 
+def hash_parse_lookup(uniq_hash):
+    global PARSED_PROGRAMS
+    PARSED_PROGRAMS[uniq_hash]['frequency'] += 1
+    return PARSED_PROGRAMS.get(uniq_hash).get('result')
+
+
+def add_hash_parse(uniq_hash, result):
+    global PARSED_PROGRAMS
+    if len(PARSED_PROGRAMS) >= 500:
+        min_freq = min(PARSED_PROGRAMS, key=PARSED_PROGRAMS.get('frequency'))
+        min_keys = [key for key in PARSED_PROGRAMS if PARSED_PROGRAMS['frequency'] == min_freq]
```

Comment:
I fear this too will consume a lot of time (to loop over the whole list), maybe we can consider a second dictionary from freq -> hashes?